### PR TITLE
FineTuning Job SDK changes.

### PR DESCRIPTION
**Pull request by sagarsumant**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_job.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+from azure.ai.ml._restclient.v2024_01_01_preview.models import (
+    ModelProvider as RestModelProvider,
+    JobBase as RestJobBase,
+)
+from azure.ai.ml._utils.utils import camel_to_snake
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationException
+from azure.ai.ml.constants._job.finetuning import FineTuningConstants
+
+
+class FineTuningJob(Job, JobIOMixin):
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_job.py:13]: Please add @eperiment annotation to these new classes. https://msdata.visualstudio.com/Vienna/_git/sdk-cli-v2?path=/docs/SDK_preview_features.md&_a=preview&anchor=%40experimental-usage-examples

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py [hunk 2]
```diff
@@ -1479,6 +1500,21 @@ def _resolve_arm_id_for_automl_job(
...
 
+    def _resolve_arm_id_for_finetuning_job(self, job: FineTuningJob, resolver: _AssetResolver) -> FineTuningJob:
+        """Resolve arm_id for FineTuningJob.
+
+        :param job: The AutoML job
+        :type job: AutoMLJob
+        :param resolver: The asset resolver function
+        :type resolver: _AssetResolver
+        :param inside_pipeline: Whether the job is within a pipeline
+        :type inside_pipeline: bool
```
Comment (banibrata-de) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:1511]: these needs update

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_vertical.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        """Set the model to be fine-tuned.
+
+        :param value: Input object representing the mlflow model to be fine-tuned.
+        :type value: typing.Union[typing.Dict, TabularLimitSettings]
+        :raises ValidationException: Expected a mlflow model input.
+        """
+        if isinstance(value, Input) and cast(Input, value).type == "mlflow_model":
+            self._model = value
+        else:
+            if not isinstance(value, dict):
```
Comment (banibrata-de) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_vertical.py:75]: curious why another if condition inside else, what is expected when   isinstance(value, dict) is true?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/azure_openai_finetuning_job.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+from azure.ai.ml.entities._job._input_output_helpers import from_rest_data_outputs, to_rest_data_outputs
+from typing import Any, Dict
+from azure.ai.ml.entities._job.finetuning.finetuning_vertical import FineTuningVertical
+from azure.ai.ml.entities._job.finetuning.azure_openai_hyperparameters import AzureOpenAIHyperparameters
+from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationException
+
+
+class AzureOpenAIFineTuningJob(FineTuningVertical):
```
Comment (banibrata-de) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/azure_openai_finetuning_job.py:16]: all the classes inside entities which are Public in pythonic way, should have  @eperiment annotation. https://msdata.visualstudio.com/Vienna/_git/sdk-cli-v2?path=/docs/SDK_preview_features.md&_a=preview&anchor=%40experimental-usage-examples

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_vertical.py (hunk 3 above)
Comment (sagarsumant) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_vertical.py:75]: I think it was unintentional, corrected it, thanks for catching it.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/azure_openai_finetuning_job.py (hunk 4 above)
Comment (sagarsumant) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/azure_openai_finetuning_job.py:16]: Updated.

**Review by banibrata-de — APPROVED** [review]

**Pull request merged**